x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/api/rules/bulk_actions/fetch_rules_by_query_or_ids.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { RulesClient } from '@kbn/alerting-plugin/server';
+import { BadRequestError } from '@kbn/securitysolution-es-utils';
+import { MAX_RULES_TO_UPDATE_IN_PARALLEL } from '../../../../../../../common/constants';
+import type { PromisePoolOutcome } from '../../../../../../utils/promise_pool';
+import { initPromisePool } from '../../../../../../utils/promise_pool';
+import type { RuleAlertType } from '../../../../rule_schema';
+import { readRules } from '../../../logic/crud/read_rules';
+import { findRules } from '../../../logic/search/find_rules';
+import { MAX_RULES_TO_PROCESS_TOTAL } from './route';
+
+export const fetchRulesByQueryOrIds = async ({
```

Comment:
nit: This function should be split into `fetchRulesByIds` and `fetchRulesByQuery` since implementation doesn't have anything in common.

Reply:
I think it's fine to keep those methods together. They encapsulate two different methods for retrieving rules based on the input and switch logic. From a business logic standpoint, it's a single operation: rule retrieval. How the rules are retrieved is just a detail of implementation.